src/Functions/lgamma.cpp

```diff
@@ -1,7 +1,7 @@
 #include <Functions/FunctionMathUnary.h>
 #include <Functions/FunctionFactory.h>
 
-#ifndef lgamma_r
+#if defined(OS_DARWIN)
```

Comment:
I think it's Ok regardless of platform.
Function prototypes (declarations) can be repeated multiple times.

Reply:
But build for gcc is broken

Reply:
```
../src/Functions/lgamma.cpp:7:12: error: redundant redeclaration of 'double lgamma_r(double, int*)' in same scope [-Werror=redundant-decls]
```

https://clickhouse-builds.s3.yandex.net/13842/0b703260f419b5d027f49dd9b248142b8cc39276/clickhouse_build_check/build_log_751187427_1597756584.txt